src/components/accordion-group/accordion/bl-accordion.ts

```diff
@@ -0,0 +1,176 @@
+import { html, LitElement, PropertyValues, TemplateResult } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+import { CSSResultGroup } from "lit/development";
+import { classMap } from "lit/directives/class-map.js";
+import { event, EventDispatcher } from "../../../utilities/event";
+import { stringBooleanConverter } from "../../../utilities/string-boolean.converter";
+import "../../icon/bl-icon";
+import { BaklavaIcon } from "../../icon/icon-list";
+import style from "./bl-accordion.css";
+
+// TODO : COMMENTS
+@customElement("bl-accordion")
+export default class BlAccordion extends LitElement {
+  /**
+   * Whether the accordion is expanded
+   */
+  @property({ reflect: true, type: Boolean })
+  open = false;
+
+  /**
+   * Sets accordion caption.
+   */
+  @property({ reflect: true })
+  caption?: string;
+
+  /**
+   * Add icon to beginning of the title
+   */
+  @property({ converter: stringBooleanConverter() })
+  icon?: boolean | BaklavaIcon;
+
+  /**
+   * Whether the accordion is disabled
+   */
+  @property({ reflect: true, type: Boolean })
+  disabled = false;
+
+  /**
+   * Fires when accordion open state change.
+   */
+  @event("bl-toggle") private _onToggle: EventDispatcher<boolean>;
+
+  @property({ type: Number })
+  animationDuration = 100;
+
+  private _animation: Animation | null;
+  private _isExpanding = false;
+  private _isCollapsing = false;
+
+  @query("details")
+  detailsEl: HTMLDetailsElement;
+
+  @query("summary")
+  summaryEl: HTMLElement;
+
+  @query(".accordion-content")
+  contentEl: HTMLElement;
+
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  _animate(isOpening: boolean) {
+    if (isOpening) {
```

Comment:
We're using it in a different location. Perhaps we can convert it into a function

Reply:
changed structure